Orange/widgets/data/owcolor.py

```diff
@@ -350,6 +350,7 @@ def set_data(self, data):
                                         self._create_proxies(domain.metas))
             self.openContext(data)
             self.data = data.transform(domain)
+            self.data.domain = domain
```

Comment:
CMIIW, but changing `Table.from_table` to compare domains with `is` instead of `==` doesn't look like a trivial change. Also it by itself doesn't make `OWColor` work — the problem then arises in DomainConversion, which doesn't seem to know how to do the transformation. As a result, we get a data table with all values missing.

Before, as we checked with `==` if domains matched we called `from_table_rows` which just makes a selection of rows and keeps the provided domain. Now, if `domain1 == domain2` but `domain1 is not domain2` we need to make a domain conversion.

It seems like the core of the problem with all values missing is in how DomainConversion checks whether some variable is inside a domain. The relevant part of the code from `domain.py`:
```
    def __contains__(self, item):
        """
        Return `True` if the item (`str`, `int`, :class:`Variable`) is
        in the domain.
        """
        return item in self._indices
```
It seems like `item in self._indices == False` while `item in list(self._indices.keys()) == True`. Is this a result of how we compare variables?

Should we maybe just revert to `self.data.domain = domain` as it was before @janezd refactoring? If so, @janezd can you provide an explanation, since I'm not quite sure how variable's proxies work?